### PR TITLE
Rc for gimbal -- Refactor of Gimbal and chassis code

### PR DESCRIPTION
**Pull request by BenYKB**
**Whats new:**
Structs and functions used in gimbal and chassis were modified to be more readable

**What bugs were fixed:**
Gimbal control in master was previously broken

**What bugs were (potentially) introduced:**
Possible confusion of struct naming 

**Documentation Links:**

**Comments:**

### CODE REVIEW

File: user/TASK/gimbal_task/gimbal_task.c [hunk 1]
```diff
@@ -26,73 +26,119 @@
...
-        vTaskDelay(1);
         
-        //send_to_uart(gimbal_pitch_motor, pid_pitch, pitch_signal);  //Sending data via UART
+        CAN_CMD_GIMBAL( (int16_t) gimbal.yaw_motor.current_out, 
+                        (int16_t) gimbal.pitch_motor.current_out,
+                        0,
+                        0);
+        
+          //Sending data via UART
+        vTaskDelay(5);
```
Comment (Alice-Xiong) [inline user/TASK/gimbal_task/gimbal_task.c]: We should use defines for vTaskDelay duration. Also gimbal task delay should be 1, for purpose of shoot integration.

File: user/TASK/gimbal_task/gimbal_task.c [hunk 2]
```diff
@@ -120,28 +165,29 @@ int get_vision_signal(void) {
...
-    serial_send_string(str);       
-
-    sprintf(str, "current: %d\n\r", gimbal.yaw_motor->current_raw);
+    if(gimbal_msg->yaw_motor.pos_read == NULL){
+        sprintf(str, "null ... :( %d\n\r", 123);
     serial_send_string(str);
-
-    sprintf(str, "motor kp: %f\n\r", pid_pitch.Kp);
+    } else{
+        sprintf(str, "yaw position read: %d\n\r", 321);
```
Comment (Alice-Xiong) [inline user/TASK/gimbal_task/gimbal_task.c]: Can we delete this debug print lol

File: user/TASK/gimbal_task/gimbal_task.c [hunk 3]
```diff
@@ -26,73 +26,119 @@
...
-    fp32 pid_constants[3] = {pid_kp, pid_ki, pid_kd};
-    PidTypeDef pid_pitch;
-    PID_Init(&pid_pitch, PID_POSITION, pid_constants, max_out, max_iout);
-
-    PidTypeDef pid_yaw;
-    PID_Init(&pid_yaw, PID_POSITION, pid_constants, max_out, max_iout);
-    
-
-    rc_ptr = get_remote_control_point();
+    vTaskDelay(200);
```
Comment (Alice-Xiong) [inline user/TASK/gimbal_task/gimbal_task.c]: I see this value is defined in gimbal_task.h as 300... Hmm

File: user/TASK/gimbal_task/gimbal_task.c (hunk 1 above)
Comment (jingleton) [reply user/TASK/gimbal_task/gimbal_task.c]: Ok

File: user/TASK/gimbal_task/gimbal_task.c (hunk 2 above)
Comment (jingleton) [reply user/TASK/gimbal_task/gimbal_task.c]: Added a todo - fill as needed

File: user/TASK/gimbal_task/gimbal_task.c (hunk 3 above)
Comment (jingleton) [reply user/TASK/gimbal_task/gimbal_task.c]: Updated to 300 + define

**Pull request merged**